pkg/domain/component.go

```diff
@@ -5,19 +5,21 @@ import (
 	"fmt"
 	"github.com/Masterminds/semver/v3"
 	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/common"
+	"github.com/cloudogu/k8s-blueprint-operator/pkg/domain/ecosystem"
 )
 
 // Component represents a CES component (e.g. operators), its version, and the installation state in which it is supposed to be
 // after a blueprint was applied.
 type Component struct {
-	// TODO Add ComponentConfig for CRD fields like deployNamespace or helmValuesOverwrite.
 	// Name defines the name and namespace of the component. Must not be empty.
 	Name common.QualifiedComponentName
 	// Version defines the version of the package that is to be installed. Must not be empty if the targetState is
 	// "present"; otherwise it is optional and is not going to be interpreted.
 	Version *semver.Version
 	// TargetState defines a state of installation of this package. Optional field, but defaults to "TargetStatePresent"
 	TargetState TargetState
+	// PackageConfig defines generic properties for the component. This field is optional.
+	PackageConfig ecosystem.PackageConfig
```

Comment:
Mh you are right. `DeployConfig` sounds better. I used `PackageConfig` because i think the components will always be packages in the sense of the helm package manager or the apt package manager. But `DeployConfig` fullfills this thought too.